app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeScreen.kt

```diff
@@ -22,77 +28,131 @@ import co.nimblehq.compose.crypto.ui.theme.Style.textColor
 @Suppress("FunctionNaming", "LongMethod")
 @Composable
 fun HomeScreen() {
+    val scrollState = rememberScrollState()
+
     Surface {
-        ConstraintLayout(
-            modifier = Modifier.fillMaxSize()
-        ) {
-            val (
-                title,
-                portfolioCard,
-                myCoinsTitle,
-                seeAll,
-                myCoins
-            ) = createRefs()
-
-            Text(
-                modifier = Modifier
-                    .padding(top = Dp16)
-                    .constrainAs(title) {
-                        top.linkTo(parent.top)
-                        linkTo(start = parent.start, end = parent.end)
-                        width = Dimension.preferredWrapContent
-                    },
-                text = stringResource(id = R.string.home_title),
-                textAlign = TextAlign.Center,
-                style = Style.semiBold24(),
-                color = MaterialTheme.colors.textColor
-            )
-
-            PortfolioCard(
-                modifier = Modifier
-                    .constrainAs(portfolioCard) {
-                        top.linkTo(title.bottom, margin = Dp40)
-                    }
-                    .padding(horizontal = Dp16)
-            )
+        BoxWithConstraints {
```

Comment:
Not sure why we need to convert from `ConstraintLayout` to `BoxWithConstraints` 🤔

Reply:
@hoangnguyen92dn, As leave description `insight.` ☝️ The root layout to `BoxWithConstraints` instead of `ConstraintLayout` due to we have to support scrolling, So using ConstraintLayout doesn't work when we have a `nested scrolling` in the screen.

Reply:
🤔 It looks like a workaround instead. The BoxWithConstraints should be supposed for adaptive layout. I will take a look into this.